noun/retrieve.c

```diff
@@ -1770,113 +1518,122 @@ u3_mur_words(const c3_w* key_w, c3_w len_w)
   }
 }
 
-/* u3_mur_both():
+/* u3r_mug_bytes(): Compute the mug of `buf`, `len`, LSW first.
 **
-**   Join two murs.
+**   XX temporary, reconcile with u3r_mug_words
+*/
+c3_w
+u3r_mug_bytes(const c3_y *buf_y,
+              c3_w        len_w)
+{
+  c3_w syd_w = 0xcafebabe;
+
+  while ( 1 ) {
+    c3_w haz_w, ham_w;
+
+    MurmurHash3_x86_32(buf_y, len_w, syd_w, &haz_w);
+    ham_w = (haz_w >> 31) ^ (haz_w & 0x7fffffff);
+
+    if ( 0 != ham_w ) return ham_w;
+    else syd_w++;
+  }
+}
```

Comment:
This duplication is a little unfortunate, but we may want to keep it in. Murmur3 operates on uint32 "blocks", with some special handling for trailing bytes. We work primarily in uint32 words, and it's obviously inefficient to convert from words to bytes only to convert from bytes to words inside the hash function.